### ISSUE TITLE
Add ability to clone store

### ISSUE THREAD

**Issue by zerobias** [zerobias/effector#13]
Use case: many instances of the same app

**Comment by Kelin2025** [comment]
For problem with
```
<>
  <A />
  <A />
  <A />
</>
```
We can generate "context" (maybe random id) inside each `<A>` for effects
Then, we can add 
- `store.on(effect.some.done, ...)` - common event for all contexts 
- `store.on(effect.each.done, ...)` - for each context separately  
We can also add `effect.id` and `effect.for(id).done` for a certain context  

What are your thoughts?

**Comment by Kelin2025** [comment]
E.g. we have to _create alers_ when _todo is added_ but there are a lot of todo stores
We can add ids for stores and do:
```
todoA.on(addTodo.for(todoA).done, ...)
```
And for alerts
```
alerts.on(addTodo.some.done, ...)
```

**Comment by zerobias** [comment]
Yep, probably needs something like this. Usually, I create a short demo to understand the benefits and concerns

I also think about the closure-based solution
https://codesandbox.io/s/vmx6wxww43

**Comment by Kelin2025** [comment]
Closure-based solution looks like a bycicle.  
I'm not React expert so I just wrote a pseudocode:
https://codesandbox.io/s/myl2l5xv18

**Comment by Kelin2025** [comment]
I can help with implementation if it's OK

**Comment by zerobias** [comment]
Finally, implemented in [effector 21.0.0](https://github.com/zerobias/effector/releases/tag/effector%4021.0.0) as fork

**Comment by sergeysova** [comment]
🎉
